src/python/pycold/colds.py

```diff
@@ -0,0 +1,256 @@
+    """
+    _validate_params(func_name='sccd_detect', pos=pos, conse=conse, b_c2=b_c2)
+    dates, ts_b, ts_g, ts_r, ts_n, ts_s1, ts_s2, ts_t, qas, break_dates = _validate_data(dates, ts_b, ts_g, ts_r, ts_n,
+                                                                                         ts_s1, ts_s2, ts_t, qas,
+                                                                                         break_dates)
+
+    return _obcold_reconstruct(dates, ts_b, ts_g, ts_r, ts_n, ts_s1, ts_s2, ts_t, qas, break_dates, pos, conse, b_c2)
+
+
+def sccd_detect(dates, ts_b, ts_g, ts_r, ts_n, ts_s1, ts_s2, ts_t, qas, t_cg=15.0863, pos=1, conse=6, b_c2=False,
+                b_pinpoint=False, gate_tcg=9.236, b_monitor_init=False):
+    """
+    pixel-based offline SCCD algorithm.
+    Ye, S., Rogan, J., Zhu, Z., & Eastman, J. R. (2021). A near-real-time approach for monitoring forest
+    disturbance using Landsat time series: Stochastic continuous change detection.
+    Remote Sensing of Environment, 252, 112167.
+
+    Parameters
+    ----------
+    dates: 1d array of shape(observation numbers), list of ordinal dates
+    ts_b: 1d array of shape(observation numbers), time series of blue band.
+    ts_g: 1d array of shape(observation numbers), time series of green band
+    ts_r: 1d array of shape(observation numbers), time series of red band
+    ts_n: 1d array of shape(observation numbers), time series of nir band
+    ts_s1: 1d array of shape(observation numbers), time series of swir1 band
+    ts_s2: 1d array of shape(observation numbers), time series of swir2 band
+    ts_t: 1d array of shape(observation numbers), time series of thermal band
+    qas: 1d array, the QA cfmask bands. '0' - clear; '1' - water; '2' - shadow; '3' - snow; '4' - cloud
+    t_cg: threshold of change magnitude, default is chi2.ppf(0.99,5)
+    pos: position id of the pixel, i.e.,  (row -1) * ncols + col, row and col starts from 1
+    conse: consecutive observation number
+    b_c2: bool, a temporal parameter to indicate if collection 2. C2 needs ignoring thermal band for valid pixel
+                test due to its current low quality
+    b_pinpoint: bool, output pinpoint break where pinpoint is an overdetection of break using conse =3
+                        and threshold = gate_tcg, which are used to simulate the situation of NRT scenario and
+                        for training a machine-learning model
+    gate_tcg: the gate change magnitude threshold for defining anomaly
+    b_monitor_init: bool, output change metrics during the initialization stage
+    Note that passing 2-d array to c as 2-d pointer does not work, so have to pass separate bands
+    Returns
+    ----------
+    SccdOutput: the SCCD outputs that characterizes each temporal segment if b_pinpoint==False
+    Or
+    [SccdOutput, SccdReccgPinpoint]: b_pinpoint==True
+    """
+    _validate_params(func_name='sccd_detect', t_cg=t_cg, pos=pos, conse=conse, b_c2=b_c2, b_pinpoint=b_pinpoint,
```

Comment:
These function signatures have a lot of repeated code in them. Packing and unpacking the long list of parameters multiple times seems rather cumbersome. I'm not sure if I have a suggestion to make it easier, it's just something that's jumping out at me.

I think writing the C functions in this way makes sense, because its reasonably efficient, but I'm wondering if it might be better on the Python side (and maybe even on the C-side) to have some sort of structure that encapsulates the date and band arrays in a single object.

Right now it would be very easy to make a typo and say flip the ts_s1 and ts_s2 bands, and it would be difficult to check that because there are so many places where they are packed / unpacked.

Having a data structure that encapsulates the arrays of bands might also make it easier to extend this algorithm to other sensors with different spectral properties.

Reply:
Thank you for the comment. I actually was inspired by the interface function design from [pyccd](https://github.com/repository-preservation/lcmap-pyccd/blob/2eaf9933d76b861fc243327b323b80fbcf301932/ccd/__init__.py#L121). Since there has been a wide body of users for pyccd, I hoped the interface function of pycold could be the same as pyccd. I guess that their reason for using unpacked bands is that CCDC/COLD is a band-dependent algorithm: each band has different roles in the algorithm, such as Tmask (a component of COLD) used green and sr1. So making the band explicit in the inputs may make it easier and more straightforward for users to check or use other band source through checking/controlling the inputted parameters. But maybe this way makes it more difficult?

Reply:
I'm sure it's a tradeoff. It's just something that came to mind when I was reading the code.

The docs are good, so it's clear what each input should be.